support/cas-server-support-duo-core/src/main/java/org/apereo/cas/adaptors/duo/authn/DuoAuthenticationHandler.java

```diff
@@ -107,15 +104,7 @@ private DuoSecurityAuthenticationService getDuoAuthenticationService() {
         if (requestContext == null) {
             throw new IllegalArgumentException("No request context is held to locate the Duo authentication service");
         }
-        final Collection<String> providerIds = WebUtils.getResolvedMultifactorAuthenticationProviders(requestContext);
-        final Collection<MultifactorAuthenticationProvider> providers =
-            MultifactorAuthenticationUtils.getMultifactorAuthenticationProvidersByIds(providerIds,
-                ApplicationContextProvider.getApplicationContext());
-
-        if (providers.isEmpty()) {
-            throw new IllegalArgumentException("No multifactor providers are found in the current request context");
-        }
-        final MultifactorAuthenticationProvider pr = providers.iterator().next();
+        final MultifactorAuthenticationProvider pr = WebUtils.getActiveMultifactorAuthenticationProvider(requestContext);
```

Comment:
This is a good starting point, but at some point, we should get this cleaned up. A handler component should not try to reach beyond its limits and deal with webflow scopes. We can clean up this bit later.